MostViewedHandler.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * @file MostViewedHandler.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2003-2023 John Willinsky
+ * Copyright (c) 2020 Ronny Bölter, Leibniz Institute for Psychology (ZPID)
+ *
+ * Distributed under the GNU GPL v3. For full terms see the file LICENSE.
+ *
+ * @class MostViewedHandler
+ *
+ * @ingroup plugins_generic_mostViewed
+ *
+ * @brief Class for cron job functions.
+ */
+
+namespace APP\plugins\generic\mostViewed;
+
+use APP\core\Application;
+use APP\core\Services;
+use APP\facades\Repo;
+use DateTime;
+use PKP\plugins\PluginRegistry;
+use PKP\scheduledTask\ScheduledTask;
+
+class MostViewedHandler extends ScheduledTask
+{
+    public const MOST_READ_DAYS = 30;
+    public const DEFAULT_AMOUNT = 5;
+    public const MAX_YEARS_BACK = 5;
+
+    /**
+     * @copydoc ScheduledTask::getName()
+     */
+    public function getName(): string
+    {
+        return __('admin.scheduledTask.mostViewed');
+    }
+
+    /**
+     * This function is called via cron job or acron plugin.
+     *
+     * @copydoc ScheduledTask::executeActions()
+     */
+    public function executeActions(): bool
+    {
+        /** @var MostViewedPlugin $plugin */
+        $plugin = PluginRegistry::getPlugin('generic', 'mostviewedplugin');
+        if (!$plugin->getEnabled()) {
+            return false;
+        }
+        $contextDao = Application::getContextDAO();
+        foreach ($contextDao->getAll()->toIterator() as $context) {
+            if (!$plugin->getEnabled($context->getId())) {
+                continue;
+            }
+            // Default Settings for each Journal
+            $mostReadDays = static::MOST_READ_DAYS;
+            $amount = static::DEFAULT_AMOUNT;
+            $maxYearsBack = static::MAX_YEARS_BACK;
+            // Overwrite Settings if Journal has Settings
+            $settings = $plugin->getSetting($context->getId(), 'settings');
+            if ($settings) {
+                $settings = json_decode($settings, true);
+                if (is_numeric($settings['days'] ?? null) && intval($settings['days']) > 0) {
```

Comment:
Here's the only problem that I had with the previous code.

It was using `is_int()` instead of `is_numeric()`, then a string like `'123'` wasn't being recognized as a number.